controllers/argocd/sso.go

```diff
@@ -104,6 +105,13 @@ func verifyTemplateAPI() error {
 
 func (r *ReconcileArgoCD) reconcileSSO(cr *argoprojv1a1.ArgoCD) error {
 	if cr.Spec.SSO.Provider == argoprojv1a1.SSOProviderTypeKeycloak {
+		if !reflect.DeepEqual(cr.Spec.Dex, argoprojv1a1.ArgoCDDexSpec{}) {
```

Comment:
This condition wont work :( as the structure contains non-comparable objects. We need to break down it to something like this.

if cr.Spec.Dex.OpenShiftOAuth == true || cr.Spec.Dex.Config != "" {
......
}